src/Types/Exception/TypeArgumentCountError.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types\Exception;
+
+use ArgumentCountError;
+use Exception;
+
+use function sprintf;
+
+/** @psalm-immutable */
```

Comment:
We switched all Doctrine projects to use PHPStan annotations.
```suggestion
/** @immutable */
```

But I'm not sure this is necessary for this class. It doesn't have public property.